### PR TITLE
BaSP-MR-29: MEMBER TEST - get, getById and create

### PR DESCRIPTION
**Pull request by AleMac17**
Seed
Create one member in member.js seed for the unit tests

Unit Test
Create test for get method.
Create test for getById method.
Create test for create method.

### CODE REVIEW

File: src/tests/members.test.js [hunk 1]
```diff
@@ -1,5 +1,125 @@
...
+    expect(response.status).toBe(200);
+    /* eslint no-underscore-dangle: 0 */
+    expect(response.body.data._id).toBe(mockMemberSeed.id);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.message).toBe('Member found successfully!');
+  });
+});
+
+describe('post /api/members', () => {
+  test('should return status 2010', async () => {
```
Comment (AleMac17) [inline src/tests/members.test.js:93]: This is status 201

File: src/tests/members.test.js [hunk 2]
```diff
@@ -1,5 +1,141 @@
...
+    expect(response.error).toBeTruthy();
+  });
+});
+
+describe('GETById /api/members/:id', () => {
+  test('should return status 400, send no id', async () => {
+    const response = await request(app).get('/api/members/:id').send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
```
Comment (JuanManuel-Blois) [inline src/tests/members.test.js:66]: We also need to check the response message

File: src/tests/members.test.js [hunk 3]
```diff
@@ -1,5 +1,141 @@
...
+  });
+  test('should return status 400, send invalid id', async () => {
+    const response = await request(app).get(`/api/members/${mockMemberBadValidations.phone}`).send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
```
Comment (JuanManuel-Blois) [inline src/tests/members.test.js:72]: We also need to check the response message

**Review by JuanManuel-Blois — APPROVED** [review]

File: src/tests/members.test.js [hunk 4]
```diff
@@ -1,5 +1,151 @@
...
+    expect(response.body.error).toBeTruthy();
+  });
+});
+
+describe('GETById /api/members/:id', () => {
+  test('should return status 400, send no id', async () => {
+    const response = await request(app).get('/api/members/:id').send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
```
Comment (nicolasblasco) [inline src/tests/members.test.js:66]: Should we test data.length when the data is an object? Maybe we could do
`expect(response.body.data).toBeEqual({});`

File: src/tests/members.test.js [hunk 5]
```diff
@@ -1,5 +1,151 @@
...
+    expect(response.body.error).toBeTruthy();
+    expect(response.status).toBe(404);
+    expect(response.body.data.length).toBe(0);
+    expect(response.body.message).toBe('There are no members!');
+  });
+  test('should respond with a 500 status, server error', async () => {
+    jest.spyOn(Member, 'find').mockRejectedValue(new Error('Something went wrong'));
+    const response = await request(app).get('/api/members/').send();
+    expect(response.status).toBe(500);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/tests/members.test.js:57]: We should also test the message and data of the body

File: src/tests/members.test.js [hunk 6]
```diff
@@ -1,5 +1,151 @@
...
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
+    expect(response.body.message).toBe('This is not a valid object Id');
+  });
+  test('should return status 400, send invalid id', async () => {
+    const response = await request(app).get(`/api/members/${mockMemberBadValidations.phone}`).send();
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
```
Comment (nicolasblasco) [inline src/tests/members.test.js:73]: Same as above

File: src/tests/members.test.js [hunk 7]
```diff
@@ -1,5 +1,151 @@
...
+    expect(response.body.message).toBe('This is not a valid object Id');
+  });
+  test('should return status 404, send valid id but without member', async () => {
+    const response = await request(app).get(`/api/members/${mockIdBad.id}`).send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.data.length).toBe(undefined);
```
Comment (nicolasblasco) [inline src/tests/members.test.js:80]: Same as above

File: src/tests/members.test.js [hunk 8]
```diff
@@ -1,5 +1,151 @@
...
+    const response = await request(app).post('/api/members/').send(mockMemberGood);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.message).toBe('Member could not be created!');
+  });
+  test('should respond with a 500 status, server error', async () => {
+    jest.spyOn(Member, 'create').mockRejectedValue(new Error('Something went wrong'));
+    const response = await request(app).post('/api/members/').send(mockMemberGood);
+    expect(response.status).toBe(500);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/tests/members.test.js:149]: We should also test the message and data of the body

File: src/tests/members.test.js [hunk 9]
```diff
@@ -1,5 +1,151 @@
...
+    expect(response.body.data).toBe(undefined);
+    // eslint-disable-next-line
+    expect(response.body.message).toBe('There was an error in the validation: \"firstName\" must be a string');
+  });
+  test('should respond with a 400 status, server error', async () => {
+    jest.spyOn(Member, 'create').mockResolvedValue(null);
+    const response = await request(app).post('/api/members/').send(mockMemberGood);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.message).toBe('Member could not be created!');
```
Comment (nicolasblasco) [inline src/tests/members.test.js:143]: We should also test the data of the body

File: src/tests/members.test.js [hunk 10]
```diff
@@ -1,5 +1,151 @@
...
+    /* eslint no-underscore-dangle: 0 */
+    expect(response.body.data._id).toBe(mockMemberSeed.toString());
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.message).toBe('Member found successfully!');
+  });
+  test('should respond with a 500 status, server error', async () => {
+    jest.spyOn(Member, 'findById').mockRejectedValue(new Error('Something went wrong'));
+    const response = await request(app).get(`/api/members/${mockMemberSeed}`).send();
+    expect(response.status).toBe(500);
+    expect(response.body.error).toBeTruthy();
```
Comment (nicolasblasco) [inline src/tests/members.test.js:95]: We should also test the message and data of the body

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

**Review by LucianoAlarcon — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**